File: packages/backend/dict-simulator/src/models/PixKey.ts
import mongoose, { Schema, Document } from 'mongoose'

export enum KeyType {
  CPF = 'CPF',
  CNPJ = 'CNPJ',
  EMAIL = 'EMAIL',
  PHONE = 'PHONE',
  RANDOM = 'RANDOM',
}

export enum AccountType {
  CHECKING = 'CHECKING',
  SAVINGS = 'SAVINGS',
}

export enum KeyStatus {
  ACTIVE = 'ACTIVE',
  BLOCKED = 'BLOCKED',
  FROZEN = 'FROZEN',
}

export interface IPixKey extends Document {
  key: string
  keyType: KeyType
  accountType: AccountType
  ispb: string
  branch: string
  accountNumber: string
  accountHolderName: string
  accountHolderDoc: string
  status: KeyStatus
  createdAt: Date
  updatedAt: Date
}

const PixKeySchema = new Schema<IPixKey>(
  {
    key: {
      type: String,
      required: true,
      unique: true,
      trim: true,
    },
    keyType: {
      type: String,
      enum: Object.values(KeyType),
      required: true,
    },
    accountType: {
      type: String,
      enum: Object.values(AccountType),
      required: true,
    },
    ispb: {
      type: String,
      required: true,
    },
    branch: {
      type: String,
      required: true,
    },
    accountNumber: {
      type: String,
      required: true,
    },
    accountHolderName: {
      type: String,
      required: true,
    },
    accountHolderDoc: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: Object.values(KeyStatus),
      default: KeyStatus.ACTIVE,
    },
  },
  {
    timestamps: true,
  },
)

export const PixKey = mongoose.model<IPixKey>('PixKey', PixKeySchema)

File: packages/backend/dict-simulator-go/internal/models/pixkey.go
package models

import "time"

type KeyType string

const (
	KeyTypeCPF    KeyType = "CPF"
	KeyTypeCNPJ   KeyType = "CNPJ"
	KeyTypeEmail  KeyType = "EMAIL"
	KeyTypePhone  KeyType = "PHONE"
	KeyTypeRandom KeyType = "RANDOM"
)

type AccountType string

const (
	AccountTypeChecking AccountType = "CHECKING"
	AccountTypeSavings  AccountType = "SAVINGS"
)

type KeyStatus string

const (
	KeyStatusActive  KeyStatus = "ACTIVE"
	KeyStatusBlocked KeyStatus = "BLOCKED"
	KeyStatusFrozen  KeyStatus = "FROZEN"
)

type PixKey struct {
	ID                string        `json:"id"`
	Key               string        `json:"key"`
	KeyType           KeyType       `json:"keyType"`
	AccountType       AccountType   `json:"accountType"`
	ISPB              string        `json:"ispb"`
	Branch            string        `json:"branch"`
	AccountNumber     string        `json:"accountNumber"`
	AccountHolderName string        `json:"accountHolderName"`
	AccountHolderDoc  string        `json:"accountHolderDoc"`
	Status            KeyStatus     `json:"status"`
	CreatedAt         time.Time     `json:"createdAt"`
	UpdatedAt         time.Time     `json:"updatedAt"`
}
